addon/utils.js

```diff
@@ -195,3 +195,30 @@ export function parseVariables (valueObj, queryJSON, startPath = '') {
 export function populateQuery (valueObj, query, startPath = '') {
   return JSON.parse(parseVariables(valueObj, JSON.stringify(query), startPath))
 }
+
+/**
+ * Checks if the query object has valid values
+ * @param {Object} value - form value
+ * @param {Object} queryDef - query definition
+ * @param {String} startPath - starting path
+ * @returns {Boolean} true if valid
+ */
+export function hasValidQueryValues (value, queryDef, startPath) {
+  if (!queryDef) {
+    return true
+  }
+
+  const query = populateQuery(value, queryDef, startPath)
+  return Object.keys(query).every((key) => {
+    const prop = query[key]
+
+    // query contains multiple key-value comma-delimited pairs
+    if (prop && (prop.indexOf(':') >= 0 || prop.indexOf(',') >= 0)) {
+      return prop.split(',').every((q) => {
+        return !_.isEmpty(q.split(':')[1])
+      })
+    }
```

Comment:
I'm a little concerned about this implementation. We're basically saying here that the values being pulled from somewhere else in the value object have to be used as the value of some part of the query, and we're handling the special case of the `:` and `,` delimited format for certain APIs we interface with. I think it would be more robust if we just checked that all the values references using the `${}` notation were present in the value object. 


What if one of those values is used as the property name, instead of the property value, in the query? This solution make some assumptions about where the `${}` syntax can be used that I don't think we had before (or at least I don't think we want to have). I don't think we should need to, or want to handle the `:` and `,` syntax specially at all, anywhere. If we can come up with a general solution, it should work in that special case too, and doesn't require us to have seemingly random special cases in our code :)